### DISCUSSION TITLE
SHA256 mismatches for artifacts downloaded from GitHub archive links?

### DISCUSSION THREAD

**Discussion by cameronbriar** [community/community#45832] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

For example: https://github.com/kubernetes-sigs/krew-index/blob/master/plugins/ns.yaml

The artifact (https://github.com/ahmetb/kubectx/archive/v0.9.4.tar.gz) no longer matches the described checksum (`91e6b2e0501bc581f006322d621adad928ea3bd3d8df6612334804b93efd258c`).

I've seen several different instances of this as of a few hours ago.

Did something change server-side? (e.g. compression algo, etc.) Why are downloaded artifacts from `https://github.com/XXX/YYY/archive/ZZZ.tar.gz` no longer the same?

**Comment by yann-morin-1998** [comment]
See also #45830

**Comment by cameronbriar** [comment, marked as answer]
Answered: https://github.blog/changelog/2023-01-30-git-archive-checksums-may-change/

**Comment by cameronbriar** [comment]
Anticipating a revert given: https://github.com/bazel-contrib/SIG-rules-authors/issues/11#issuecomment-1409438954
